python/pyarrow/_flight.pyx

```diff
@@ -1871,7 +1909,6 @@ cdef CStatus _server_authenticate(void* self, CServerAuthSender* outgoing,
         reader.poison()
```

Comment:
Is this a typo "poison"?

Reply:
Not a typo.

Reply:
This is to ensure a server doesn't use the Python reader beyond the lifetime of the C++ reader it wraps.